cstar/base/scheduler_job.py

```diff
@@ -0,0 +1,287 @@
+## STIL TODO:
```

Comment:
I have now restructured things so there's a dedicated `system` directory.
However `scheduler.py` and `environment.py` are sort of special cases in that they _define_ universal aspects of the system that should be available everywhere else in the package (via `CStarSystemManager`). As such, these are the only two modules that do not import from anywhere else in C-Star (to avoid circular importing). 

While seemingly closely related to `scheduler.py`, `scheduler_job.py` defines HPC job management-related methods, and should be able to import from elsewhere in C-Star (e.g. it requires access to system-level environment information, rather than providing this information). While on the surface `scheduler.py` and `scheduler_job.py` may seem related, this distinction means they should be kept separate.

The prefix is necessary to avoid ambiguity, as C-Star can run jobs without a scheduler.

Reply:
I could do `cstar/system/scheduler.py` and `cstar/scheduler/job.py`? I think that would perhaps even make more sense.  May just seem a bit more fragmented, e.g. `job.py` would be the only module in `scheduler/`.

Reply:
Happy with that also. We can also easily change this later as long as it's not too tangled.

Reply:
I can do it now, no sweat

Reply:
> I can do it now, no sweat

done in 5aef23d08166d293be4b81f23819dd11bd02755d and c05de1f3598a7a558e5d557c9fe69ddd8e6a1d72